src/compiler/transformers/es2015.ts

```diff
@@ -1024,8 +1024,11 @@ namespace ts {
                 }
             }
 
-            // Return the result if we have an immediate super() call on the last statement.
-            if (superCallExpression && statementOffset === ctorStatements.length - 1) {
+            // Return the result if we have an immediate super() call on the last statement,
+            // but only if the constructor itself doesn't use 'this' elsewhere.
+            if (superCallExpression
+            && statementOffset === ctorStatements.length - 1
+            && !(ctor.transformFlags & (TransformFlags.ContainsLexicalThis | TransformFlags.ContainsCapturedLexicalThis))) {

```

Comment:
So `ContainsLexicalThis` means merely that a `this` keyword was somewhere inside of the function. `ContainsCapturedLexicalThis` means that a `this` keyword was captured by an arrow function inside the function.